### PR TITLE
Update to 2022.6.21

### PR DESCRIPTION
**Pull request by cddjr**
Based on v2022.6.22e(2108)

### CODE REVIEW

**Review by DarkPenguin24 — DISMISSED** [review]

**Comment by cddjr** [comment]
> > 基于 v2022.6.22e(2108)
> 
> Error code c2760 Syntax error: unexpected token ')'

<details>
<summary>git pull --force </summary>
估计你拉取的是老代码了，你再重新强制拉取一次
</details>

**Comment by demonitize** [comment]
Any ETA on the build being uploaded

File: appdata/il2cpp-types.h [hunk 1]
```diff
@@ -4588,24 +4588,23 @@ namespace app
...
-        void* PlayerIcon;
-        void* Background;
-        void* MaskArea;
-        void* Flag;
-        void* Megaphone;
-        void* Overlay;
-        void* XMark;
-        void* GAIcon;
-        void* ThumbsDown;
+        struct UiElement* ConfirmButton;
```
Comment (kotae4) [inline appdata/il2cpp-types.h:4596]: UiElement is not defined anywhere. Stick with void* if you don't want to include definitions and don't need to use the field.

File: appdata/il2cpp-types.h [hunk 2]
```diff
@@ -4614,12 +4613,12 @@ namespace app
         uint8_t VotedFor;
         bool voteComplete;
         bool resultsShowing;
+        struct PoolablePlayer* PlayerIcon;
```
Comment (kotae4) [inline appdata/il2cpp-types.h:4616]: PoolablePlayer is not defined. See comment about UiElement

File: appdata/il2cpp-types.h [hunk 3]
```diff
@@ -4873,49 +4872,48 @@ namespace app
...
         struct Color CameraColor;
         float MaxLightRadius;
         float MinLightRadius;
         float MapScale;
-        void* MapPrefab;
-        void* ExileCutscenePrefab;
-        void* EmergencyOverlay;
-        void* ReportOverlay;
-        void* MeetingBackground;
+        struct MapBehaviour* MapPrefab;
```
Comment (kotae4) [inline appdata/il2cpp-types.h:4881]: MapBehaviour not defined

File: appdata/il2cpp-types.h [hunk 4]
```diff
@@ -4588,24 +4588,23 @@ namespace app
 #pragma endregion
```
Comment (kotae4) [inline appdata/il2cpp-types.h:4588]: A lot of undefined structs being pointed at. I didn't mark them all.

File: hooks/InnerNetClient.cpp [hunk 5]
```diff
@@ -95,7 +95,7 @@ void dInnerNetClient_Update(InnerNetClient* __this, MethodInfo* method)
         if (State.originalName == "-") {
             app::GameData_PlayerOutfit* outfit = GetPlayerOutfit(GetPlayerData(*Game::pLocalPlayer));
             if (outfit != NULL)
-                State.originalName = convert_from_string(outfit->fields._playerName);
+                State.originalName = convert_from_string(outfit->fields.postCensorName);
```
Comment (kotae4) [inline hooks/InnerNetClient.cpp:98]: Should we use the postCensorName here or is there a better alternative?

File: user/utility.cpp [hunk 6]
```diff
@@ -563,7 +564,7 @@ std::string ToString(__maybenull PlayerControl* player) {
 std::string ToString(__maybenull GameData_PlayerInfo* data) {
 	if (data) {
 		if (const auto outfit = GetPlayerOutfit(data))
-			return std::format("<#{} {}>", +data->fields.PlayerId, convert_from_string(outfit->fields._playerName));
+			return std::format("<#{} {}>", +data->fields.PlayerId, convert_from_string(outfit->fields.postCensorName));
```
Comment (kotae4) [inline user/utility.cpp:568]: Is there an alternative to postCensorName? Would using it here mess with places where we change the player name (like shapeshifting or impersonation)?
Same with all the other usages of postCensorName in this file.

**Review by kotae4 — CHANGES_REQUESTED** [review]
Please switch back to void pointers in il2cpp-types for all the undeclared structs & confirm that usage of postCensorName doesn't interfere with other systems

File: appdata/il2cpp-types.h (hunk 4 above)
Comment (cddjr) [reply appdata/il2cpp-types.h:4588]: I don't think it's necessary to define these structs.

File: hooks/InnerNetClient.cpp (hunk 5 above)
Comment (cddjr) [reply hooks/InnerNetClient.cpp:98]: using postCensorName is the only way.(Decompiled from GameData_PlayerInfo$$get_PlayerName)

File: user/utility.cpp (hunk 6 above)
Comment (cddjr) [reply user/utility.cpp:568]: A better way is to call GameData_PlayerOutfit::get_PlayerName(), but it's in nested class, get_method() can't resolve the nested class yet.

**Review by manianac — APPROVED** [review]
Looks good to me

**Pull request merged**